common/OpTestLpar.py

```diff
@@ -685,3 +685,79 @@ def lpar_hexdump(self, i_dev):
             l_msg = "hexdump failed for device %s" % i_dev
             print l_msg
             raise OpTestError(l_msg)
+
+    ##
+    # @brief It will clone latest skiboot git repository in i_dir directory
+    #
+    # @param i_dir @type string: directory where skiboot source will be cloned
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def lpar_clone_skiboot_source(self, i_dir):
```

Comment:
Yes you are correct, adding functions for installing software tools is not a good idea. But we already added function for linux git tree. similarly we will add function for skiboot git repository, which will be useful for most of OPAL test cases. For remaining software tools and utilities we already have function  lpar_check_command(i_cmd) in which case test will fail, if that utility is not installed by user manually. So we will restrict to only these linux and skiboot repositories. for any other executable binary's  and utilities we will use that function lpar_check_command(i_cmd) to check whether the utility is installed or not. For EX: lpar_check_command("git") etc..